.github/workflows/pr-labeler.yml

```diff
@@ -10,8 +10,16 @@ jobs:
     steps:
     - uses: actions/labeler@v3
       with:
+        configuration-path: ".github/actions-labeler.yml"
         repo-token: "${{ secrets.GITHUB_TOKEN }}"
         sync-labels: true # add/remove labels as modified paths in the PR change
+  branches_label:
+    runs-on: ubuntu-latest
+    name: Label base branches
+    steps:
+    - uses: srvaroa/labeler@v0.9
```

Comment:
do we trust this? we are giving it credentials.

Reply:
it doesn't seem widely used: https://github.com/srvaroa/labeler

Reply:
That's a good point, though it won't get access to secrets when running on PRs from folks without repo write access.

The code of the action is pretty simple -- code review?

Reply:
The safest thing to do is to run the actions in a `pull_request` trigger, write out the full set of labels to apply as a workflow output, then trigger a second job with write access that just takes the trusted "labels to apply" data and applies them.

Reply:
i'm not worried about the code today. a lot of recent attacks have been supply chain attacks that target dependencies that are not pinned. i don't quite understand if this is pinned or not.

Reply:
i also don't know how secrets work here. i just see use passing a secret to this code.

Reply:
i'd say i don't understand this infrastructure well enough gauge risk, but brought it up hoping someone else could.

Reply:
> i'd say i don't understand this infrastructure well enough gauge risk, but brought it up hoping someone else could.

👍 I'm reasonably familiar with how this all works since I've done a few reviews recently.

Helio did correctly pin the action to v0.9 in this change (good on you btw, Helio). However, since v0.9 is a tag, it could be changed in the future. It's safest to pin to the commit hash. @heliocliu I just pushed an update to your PR directly to pin these actions using hashes, and I filed #9608 to track pinning other actions we use.

I think with the pinning we are in good shape, and can merge this.